### ISSUE TITLE
Stopping Amplify Pipeline build on failures

### ISSUE THREAD

**Issue by dhruvbansal2** [aws-amplify/amplify-cli#11344]
> **Note**: If your question is regarding the AWS Amplify Console service, please log it in the
> [AWS Amplify Console repository](https://github.com/aws-amplify/amplify-console/issues)

**Which Category is your question related to?**
Amplify Pipeline

**Amplify CLI Version**
9.2.1

You can use `amplify -v` to check the amplify cli version on your system

**What AWS Services are you utilizing?**
AWS Amplify, OpenSearch

**Provide additional details e.g. code snippets. _Be sure to remove any sensitive data._**

Currently, we are utilizing the `@searchable` directive in combination with the `override.ts` file to override OpenSearch configuration such as instance type, name, fine grain access, etc. Note, we followed [this approach](https://github.com/aws-amplify/amplify-cli/issues/9063#issuecomment-1212640682) to retrieve environment variables in the override.ts file.

```
import { ApiContext } from './config-api-override';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  const env = ApiContext['environment'];
  const region = ApiContext['region'];
  const accountId = ApiContext['accountId'];

  if (env != null) {

    const searchDomainName = `amplify-opense-${env}`;
  }
}
```

This process works as expected, however, we have experienced situations where certain code changes will prevent the code in `override.ts` from executing. This ends up recreating our entire OpenSearch domain as the OpenSearch name will have changed (which is a ["Replacement"](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html#cfn-elasticsearch-domain-domainname) action). This has caused all our production services that utilize OpenSearch to go down and we are looking for ways to prevent this. 

Is there a way we can fail our Amplify build if code in the `override.ts` fails to execute. I've attempted to `process.exit(1)` in the override.ts file but that ends up recreating the OpenSearch instance.

An example of this can be found in [this other ticket](https://github.com/aws-amplify/amplify-category-api/issues/985) I created. Ideally, if I exit(1) in the `override.ts` file (if the `env` environment variable isn't set), that should fail the build before any Cloudformation changes are made.

**Comment by josefaidt** [comment]
Hey @dhruvbansal2 :wave: thanks for raising this! While this appears to be an intended design decision to skip the override if it fails, I can empathize with how this can be detrimental to workflows. If I were to add a simple override with a syntax error, upon `amplify push` the CLI will attempt and fail to build the override but proceed with the push anyway. With that I will mark this as a bug 🙂

**Comment by dhruvbansal2** [comment]
Thank you @josefaidt.

In the meantime, do you have any workaround that can help prevent this? This issue has occurred 3+ times for us (each time causing production impact).

I am currently exploring using post push api hook to somehow validate if the override file ran but not getting much success there.

**Comment by jhockett** [comment]
This has been addressed in version 10.5 and above. Thanks for reporting this issue @dhruvbansal2!
